CalibTracker/SiStripChannelGain/test/UsefullCode/TimeHistory/PlotMacro.C

```diff
@@ -19,13 +19,27 @@
 #include "PlotFunction.h"
 
 #include "DataFormats/SiStripDetId/interface/SiStripDetId.h"
-#include "DataFormats/SiStripDetId/interface/TIBDetId.h"
-#include "DataFormats/SiStripDetId/interface/TIDDetId.h"
-#include "DataFormats/SiStripDetId/interface/TOBDetId.h"
-#include "DataFormats/SiStripDetId/interface/TECDetId.h"
+
+
+//WARNING: these methods assume that the DetId is a
+//TIBDetId or TOBDetId or TIDDetId or TECDetId (and don't check this).
+//Please use the corresponding TrackerTopology methods whenever possible.
+//Code copied from TIBDetId, TOBDetId, TIDDetId, and TECDetId (see
+//git show CMSSW_9_2_3_patch2:DataFormats/SiStripDetId/interface/TIBDetId.h
+//git show CMSSW_9_2_3_patch2:DataFormats/SiStripDetId/interface/TOBDetId.h
+//git show CMSSW_9_2_3_patch2:DataFormats/SiStripDetId/interface/TIDDetId.h
+//git show CMSSW_9_2_3_patch2:DataFormats/SiStripDetId/interface/TECDetId.h)
+
+uint32_t getTIBorTOBlayer( DetId detId ) { return ((detId.rawId()>>14) & 0x7); };
+uint32_t getTIDside      ( DetId detId ) { return ((detId.rawId()>>13) & 0x3); };
+uint32_t getTIDring      ( DetId detId ) { return ((detId.rawId()>> 9) & 0x3); };
+uint32_t getTIDwheel     ( DetId detId ) { return ((detId.rawId()>>11) & 0x3); };
+uint32_t getTECside      ( DetId detId ) { return ((detId.rawId()>>18) & 0x3); };
+uint32_t getTECring      ( DetId detId ) { return ((detId.rawId()>> 5) & 0x7); };
+uint32_t getTECwheel     ( DetId detId ) { return ((detId.rawId()>>14) & 0xF); };
```

Comment:
just wondering: where did you get the start bits and masks ? I cannot see them in the original `.h`s ?